src/main/java/com/stripe/net/APIResource.java

```diff
@@ -72,38 +72,20 @@ public static void setStripeResponseGetter(StripeResponseGetter srg) {
       .create();
 
   private static String className(Class<?> clazz) {
-    String className = clazz.getSimpleName().toLowerCase().replace("$", " ");
-
-    // TODO: Delurk this, with invoiceitem being a valid url, we can't get too
-    // fancy yet.
-    if (className.equals("applepaydomain")) {
-      return "apple_pay_domain";
-    } else if (className.equals("applicationfee")) {
-      return "application_fee";
-    } else if (className.equals("bitcoinreceiver")) {
-      return "bitcoin_receiver";
-    } else if (className.equals("countryspec")) {
-      return "country_spec";
-    } else if (className.equals("ephemeralkey")) {
-      return "ephemeral_key";
-    } else if (className.equals("exchangerate")) {
-      return "exchange_rate";
-    } else if (className.equals("fileupload")) {
-      return "file";
-    } else if (className.equals("issuerfraudrecord")) {
-      return "issuer_fraud_record";
-    } else if (className.equals("orderreturn")) {
-      return "order_return";
-    } else if (className.equals("sourcetransaction")) {
-      return "source_transaction";
-    } else if (className.equals("subscriptionitem")) {
-      return "subscription_item";
-    } else if (className.equals("threedsecure")) {
-      return "three_d_secure";
-    } else if (className.equals("usagerecord")) {
-      return "usage_record";
-    } else {
-      return className;
+    // Convert CamelCase to snake_case
+    final String className = clazz.getSimpleName()
+        .replaceAll("(.)([A-Z][a-z]+)", "$1_$2")
+        .replaceAll("([a-z0-9])([A-Z])", "$1_$2")
+        .toLowerCase();
```

Comment:
Just to verify: we're pretty confident the test suite would catch a problem here (if one happened to be introduced)?

Reply:
Yes, as long as all resources are tested and the asserted URLs are correct (which should be the case, save for some deprecated resources e.g. bitcoin receivers), then any issue in this code would be detected as it is used to format endpoint URLs. I'm relatively confident that this is the case -- the first version of my code had issues with consecutive capital letters and transformed `SKU` into `s_ku`, which caused the SKU functional tests to fail.